lib/base.js

```diff
@@ -250,9 +250,46 @@ class ConnectionPool extends EventEmitter {
           return
         }
 
+        let createAttempt = 0
+
+        function create () {
```

Comment:
I wonder if this function can be brought out of the closure here to save on redefining it every time connect is called?

Reply:
Connect isn't really called regularly and it currently helps keep scope of the vars a bit more manageable (IMO)

Reply:
I do think that these functions should be extracted and parameterised to avoid nesting too deeply, and to allow easy extension of `ConnectionPool` in userland code. It doesn't look like there would be much to parameterise.

Reply:
OK - I've looked at this and it just makes scoping a bit too hard. This feature is also going to be removed in v6 because tarn has a retry strategy that can be configured.

As such we don't want to make this available on the `ConnectionPool` for userland changes and the scoping of the retry count becomes too difficult if we move the function around...

Reply:
OK - I've managed to pull it out ;)